### DISCUSSION TITLE
Green Sun socket with Tasmota

### DISCUSSION THREAD

**Discussion by bluhminga** [arendst/Tasmota#19584] [Q&A] [answered]
Hello, I have GreenSun WLAN sockets.  These are only recognized by just one app.  Now I want to install tasmota OTA.  Has anyone ever done this with this socket?

**Comment by barbudor** [comment, marked as answer]
Not being listed in the template repository is not a definitive no-go but the scales are not tipping the right way
Are you sure there is an ESP inside ?
While connected to your network, try to find the MAC address of the device
You may find that in your router's DHCP tables
Or using Windows, ping the IP address of the socket then use command `arp -a`

Then check if the MAC address match an Espressif device here: https://www.wireshark.org/tools/oui-lookup.html

**Reply by sfromis** [reply]
And even assuming that it an Espressif chip compatible with Tasmota, OTA install of Tasmota is unlikely. The norm is to crack it open and do wired flashing with an USB/serial adapter.

**Comment by bluhminga** [comment]
Thanks for the feedback. According to the MAC address it is a Tuya Smart Inc. I have already been able to switch the sockets using TuyaCloud. I'm going to try Tuya Local now.

**Reply by tynor88** [reply]
Any update on Tuya Local @bluhminga
